pkg/process/v5/writer.go

```diff
@@ -93,6 +110,18 @@ func (w writer) metadata() (*db.Metadata, error) {
 	return &metadata, nil
 }
 
+func (w writer) ProviderMetadata() *ProviderMetadata {
+	metadata := ProviderMetadata{}
```

Comment:
should probably allocate the `Providers` field such that we don't see `"Providers": nil` if no providers are configured (`"Providers": []` is preferred).